prototype_source/inductor_with_x86_cpu_as_backend_of_pt2e.rst

```diff
@@ -0,0 +1,186 @@
+            model,
+            example_inputs
+        )
+        # Note 2: For PyTorch 2.1 release, please use the API of `torch._dynamo.export` to capture the FX Graph
+        # exported_model, guards = torch._dynamo.export(
+        #     model,
+        #     *copy.deepcopy(example_inputs),
+        #     aten_graph=True,
+        # )
+
+
+Now, we will have the FX Module to be quantized.
+
+2. Apply Quantization Flow
+----------------------------
+
+After we capture the FX Module to be quantized, we will import the Backend Quantizer for X86 CPU and configure how to
+quantize the model.
+
+::
+
+    # Create X86InductorQuantizer
+    quantizer = X86InductorQuantizer()
+    quantizer.set_global(xiq.get_default_x86_inductor_quantization_config())
+
+.. note::
+
+- The default quantization configuration in ``X86InductorQuantizer`` uses 8-bits for both weights and activations, as indicated by the argument ``reduce_range=False``.
+  When Vector Neural Network Instruction is not available, the onednn backend silently chooses kernels that assume
+  `multiplications are 7-bit x 8-bit <https://oneapi-src.github.io/oneDNN/dev_guide_int8_computations.html#inputs-of-mixed-type-u8-and-s8>`_. In other words, potential
+  numeric saturation and accuracy issue may happen when running on CPU without Vector Neural Network Instruction.
+
+After we import the backend-specific Quantizer, we will prepare the model for post-training static quantization.
+``prepare_pt2e`` folds BatchNorm operators into preceding Conv2d operators, and inserts observers in appropriate places in the model.
+
+::
+
+    # PT2E Quantization flow
+    prepared_model = prepare_pt2e(exported_model, quantizer)
+
+Now, we will calibrate the ``prepared_model`` after the observers are inserted in the model.
+
+::
+
+    # We use the dummy data as an example here
+    prepared_model(*example_inputs)
+
+    # Alternatively: user can define the dataset to calibrate
+    # def calibrate(model, data_loader):
+    #     model.eval()
+    #     with torch.no_grad():
+    #         for image, target in data_loader:
+    #             model(image)
+    # calibrate(prepared_model, data_loader_test)  # run calibration on sample data
+
+Finally, we will convert the calibrated Model to a quantized Model. ``convert_pt2e`` takes a calibrated model and produces a reference quantized model.
+
+::
+
+    converted_model = convert_pt2e(prepared_model).eval()
+
+After these steps, we finished applying the quantization flow and we will get the reference quantized model.
+
+
+3. Lower into Inductor
+------------------------
+
+After we get the reference quantized model, we will further lower it into the inductor backend.
+
+::
+
+    optimized_model = torch.compile(converted_model)
+
+    # Running some benchmark
+    optimized_model(*example_inputs)
+
+
+Put all these codes together, we will have the `toy example code <https://gist.github.com/leslie-fang-intel/3f1652ef8296ce916717e938d887e86a>`_.
```

Comment:
nit: we probably don't need to link this code since it doesn't provide much additional information and in the future if we update tutorial code it's another place we need to update.

Reply:
Thanks and removed.